internal/provider/resource_apply_module.go

```diff
@@ -387,43 +444,68 @@ func (t *applyModuleResource) doRun(ctx context.Context,
 	})
 	if err != nil {
 		diags.AddError("Failed to create run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	if err = t.waitForJobCompletion(ctx, createdRun.Plan.CurrentJobID); err != nil {
 		diags.AddError("Failed to wait for plan job completion", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	plannedRun, err := t.client.Run.GetRun(ctx, &sdktypes.GetRunInput{ID: createdRun.Metadata.ID})
 	if err != nil {
 		diags.AddError("Failed to get planned run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	// If the plan fails, both plannedRun.Status and plannedRun.Plan.Status are "errored".
 	// If the plan succeeds, plannedRun.Status is "planned",
 	// while plannedRun.Plan.Status is "finished".
 	//
-	if !strings.HasPrefix(string(plannedRun.Status), "planned") {
-		diags.AddError("Plan failed", string(plannedRun.Status))
-		return diags
+	if (plannedRun.Status != sdktypes.RunPlanned) && (plannedRun.Status != sdktypes.RunPlannedAndFinished) {
+
+		// Bring in any error message(s) from the finished inner plan run.
+		innerPlanRunDiags := t.extractRunError(ctx, plannedRun)
+		diags.Append(innerPlanRunDiags...)
+		if innerPlanRunDiags.HasError() {
+			return nil, diags
+		}
+
+		diags.AddError("Plan failed with unknown error", string(plannedRun.Status))
+		return nil, diags
 	}
-	if plannedRun.Plan.Status != "finished" {
+	if plannedRun.Plan.Status != sdktypes.PlanFinished {
+
+		// Bring in any error message(s) from the finished inner plan run.
+		innerPlanRunDiags := t.extractRunError(ctx, plannedRun)
```

Comment:
Any reason this code block between 479-487 is separate from the IF condition above?

Reply:
Why would a different error message be returned for the second condition? They both appear to be an unknown error since no error was found when it attempted to extract an error from the logs.